v2/errors.go

```diff
@@ -5,19 +5,181 @@
+	code int
+	// err is a nested error which is used to form a chain of errors for better context.
+	err error
+}
+
+// Kind determines the Kind associated with an error by inspecting the chain of errors. The top-most matching Kind is
+// returned or KinkUnknown if no Kind can be determined.
+func Kind(err error) ErrKind {
+	var e CommonEdgexError
+	if !errors.As(err, &e) {
+		return KindUnknown
+	}
+	// We want to return the first "Kind" we see that isn't Unknown, because
+	// the higher in the stack the Kind was specified the more context we had.
+	if e.kind != KindUnknown || e.err == nil {
+		return e.kind
+	}
+
+	return Kind(e.err)
+}
+
+// Error creates an error message taking all nested and wrapped errors into account.
+func (ce CommonEdgexError) Error() string {
+	if ce.err == nil {
+		return ce.message
+	}
+
+	// ce.err.Error functionality gets the error message of the nested error and which will handle both CommonEdgexError
+	// types and Go standard errors(both wrapped and non-wrapped).
+	return ce.message + " -> " + ce.err.Error()
+}
+
+// DebugMessages returns a string taking all nested and wrapped operations and errors into account.
+func (ce CommonEdgexError) DebugMessages() string {
+	if ce.err == nil {
+		return ce.op + ": " + ce.message
+	}
+
+	if w, ok := ce.err.(CommonEdgexError); ok {
+		return ce.op + ": " + ce.message + " -> " + w.DebugMessages()
+	} else {
+		return ce.op + ": " + ce.message + " -> " + ce.err.Error()
+	}
+}
+
+// Message returns the first level error message without further details.
+func (ce CommonEdgexError) Message() string {
+	return ce.message
+}
+
+// Code returns the status code of this error.
+func (ce CommonEdgexError) Code() int {
+	return ce.code
+}
+
+// Unwrap retrieves the next nested error in the wrapped error chain.
+// This is used by the new wrapping and unwrapping features available in Go 1.13 and aids in traversing the error chain
+// of wrapped errors.
+func (ce CommonEdgexError) Unwrap() error {
+	return ce.err
+}
+
+// Is determines if an error is of type CommonEdgexError.
+// This is used by the new wrapping and unwrapping features available in Go 1.13 and aids the errors.Is function when
+// determining is an error or any error in the wrapped chain contains an error of a particular type.
+func (ce CommonEdgexError) Is(err error) bool {
+	switch err.(type) {
+	case CommonEdgexError:
+		return true
+	default:
+		return false
+
+	}
+}
+
+// NewCommonEdgexError creates a new CommonEdgexError with the information provided.
+func NewCommonEdgexError(kind ErrKind, message string, wrappedError error) CommonEdgexError {
+	return CommonEdgexError{
+		kind:    kind,
+		op:      addCallerInformation(),
+		message: message,
+		code:    codeMapping(kind),
+		err:     wrappedError,
+	}
+}
+
+// NewCommonEdgexError creates a new CommonEdgexError to wrap another error to record the operation stacks.
```

Comment:
Change `NewCommonEdgexError` to `NewWrapperEdgexError`

Reply:
modified